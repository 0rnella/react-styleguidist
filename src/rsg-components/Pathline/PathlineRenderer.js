import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

const styles = ({ fontFamily, fontSize, color }) => ({
	pathline: {
		fontFamily: fontFamily.monospace,
		fontSize: fontSize.small,
		color: color.light,
	},
});

export function PathlineRenderer({ classes, children }) {
	return <div className={classes.pathline}>{children}</div>;
}

PathlineRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node,
};

export default Styled(styles)(PathlineRenderer);
