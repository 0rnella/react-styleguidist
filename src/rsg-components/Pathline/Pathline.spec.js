import React from 'react';
import { PathlineRenderer } from './PathlineRenderer';

it('renderer should a path line', () => {
	const pathline = 'foo/bar';
	const actual = shallow(<PathlineRenderer classes={{}}>{pathline}</PathlineRenderer>);

	expect(actual).toMatchSnapshot();
});
