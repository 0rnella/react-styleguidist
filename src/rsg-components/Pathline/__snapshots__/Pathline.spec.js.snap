// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should a path line 1`] = `
<div>
  foo/bar
</div>
`;
